src/gleam/iterator.gleam

```diff
@@ -431,16 +431,36 @@ fn do_flatten(flattened: fn() -> Action(Iterator(a))) -> Action(a) {
 /// ```gleam
 /// > from_list([[1, 2], [3, 4]])
 /// > |> map(from_list)
-/// > |> flatten
+/// > |> from_list
+/// > |> concat
 /// > |> to_list
 /// [1, 2, 3, 4]
 /// ```
 ///
-pub fn flatten(iterator: Iterator(Iterator(a))) -> Iterator(a) {
-  fn() { do_flatten(iterator.continuation) }
+pub fn concat(iterator: Iterator(Iterator(a))) -> Iterator(a) {
```

Comment:
This was right before:

flatten: `a(a(element)) -> a(element)`
concat: `List(a(element)) -> a(element)`

So the new one is concat

Reply:
Oops sorry I was in a rush and messed it up, it should be ok now :)